### ISSUE TITLE
Some extra DML tests

### ISSUE THREAD

**Issue by vasilismsr** [microsoft/EconML#135]
We need to add two more DML tests:
1) Test that things work well when the final model has an intercept and the inputs can have any type of acceptable dimensions (currently multi-dim output and a final model with intercept is failing, because it is testing that `self._intercept` is True, but self._intercept is an array)
```python
from econml.dml import SparseLinearDMLCateEstimator
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression, LassoCV, Lasso, MultiTaskLassoCV
import numpy as np
X = np.random.normal(size=(1000, 10))
y = X[:, 0] + np.random.normal(size=(1000,))
est = SparseLinearDMLCateEstimator(model_y=MultiTaskLassoCV(cv=3), model_t=LassoCV(cv=3),
                                   model_final=MultiTaskLassoCV(cv=3))
dx = 8
est.fit(np.hstack([y.reshape(-1,1), y.reshape(-1,1)]), X[:, 0], X[:, 1:dx], X[:, dx:])
print(X[:1])
print(est.effect(X[:1, 1:dx]))
print(est.model_final.coef_)
print(est.const_marginal_effect(X[:1, 1:dx]))
```

2) Test that things work with the default models when dimensions have all types of input and the default models are used. Currently Linear and SparseLinear DML will fail with multidim treatment or outcome, because MultiTaskLassoCV needs to be used.

**Comment by vsyrgkanis** [comment]
@kbattocchi has this issue been addressed?

**Comment by kbattocchi** [comment]
Fixed (by #174 and others)
